R_program/Main.R

```diff
@@ -574,12 +578,117 @@ remove(temp)
 
 ############################# End MARKOV Weather Generator Code ##############################
 
+############################# Phenology Code ###############################
+# This code determines plant functional type phenology
+# and then scales STEPWAT2 parameters accordingly
+if(rescale_phenology){
+  source(vegetation.file)
+  setwd(db_loc)
+  growingSeason <- read.csv("InputData_GrowingSeason.csv", header = TRUE);
+  
+  # Parse growingSeason for phenology values
+  phenology <- growingSeason[grepl(".*_phenology", growingSeason[ , 1]), , ]
+  # Use regex to name the rows using the capture group (.*)
+  row.names(phenology) <- str_match(phenology[ , 1], "(.*)_phenology")[ , 2]
+  # Remove the header column now that it is reflected in the row names.
+  phenology <- phenology[ , 2:ncol(phenology)]
+  
+  # Parse growingSeason for biomass values
+  biomass <- growingSeason[grepl(".*_biomass", growingSeason[ ,1]), , ]
+  # Use regex to name the rows using the capture group (.*)
+  row.names(biomass) <- str_match(biomass[ ,1], "(.*)_biomass")[ , 2]
+  # Remove the header column now that it is reflected in the row names.
+  biomass <- biomass[ , 2:ncol(biomass)]
+  
+  # Parse growingSeason for pctlive values
+  pctlive <- growingSeason[grepl(".*_pctlive", growingSeason[ , 1]), , ]
+  # Use regex to name the rows using the capture group (.*)
+  row.names(pctlive) <- str_match(pctlive[ , 1], "(.*)_pctlive")[ , 2]
+  # Remove the header column now that it is reflected in the row names.
+  pctlive <- pctlive[ , 2:ncol(pctlive)]
+  
+  # Parse growingSeason for litter values
+  litter <- growingSeason[grepl("LITTER", growingSeason[ , 1]), , ]
+  # Remove the header. Since there is only one row we don't need to remember it.
+  litter <- litter[ , 2:ncol(litter)]
+  
+  # Parse growingSeason for growing season values.
+  seasons <- growingSeason[grepl("growing_season", growingSeason[ , 1]), , ]
+  # Remove the header. Since there is only one row we don't need to remember it.
+  seasons <- seasons[ , 2:ncol(seasons)]
+  # Turn seasons into a vector of months where plants are expected to grow. 
+  seasons <- Position(function(x) x, seasons):Position(function(x) x, seasons, right = TRUE)
+  
```

Comment:
Now I'm wondering why not just provide growing season, phenology, biomass, pctlive, and litter in seperate input csv files? It would require more input files but we wouldn't have to do all of this text parsing. 

@chaukap what do you think?

Reply:
@kpalmqui I think that's a good idea and it would allow us to remove the stringr library.

Reply:
@chaukap great. let's move these into seperate input csvs, remove text parsing, and then remove stringr.